### ISSUE TITLE
linter: `unicorn/no-single-promise-in-promise-methods` autofix wraps in [] when result is unused

### ISSUE THREAD

**Issue by varna** [oxc-project/oxc#22114]
### What version of Oxlint are you using?

1.62.0

### What command did you run?

`npm install && npx oxlint --fix repro.js`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "rules": {
    "unicorn/no-single-promise-in-promise-methods": "error"
  }
}
```

## Before

```js
async function run() {
  await Promise.all([
    new Promise((resolve) => resolve(true)),
  ]);
}
```

## After (actual)

```js
async function run() {
  [await new Promise((resolve) => resolve(true))];
}
```

## After (expected)

```js
async function run() {
  await new Promise((resolve) => resolve(true));
}
```

## Why this is wrong

The `await Promise.all([...])` is used as an expression statement — its return value is discarded. Wrapping in `[...]` allocates an array that is never read. The fixer should detect that the parent context discards the value and emit just `await <element>;`.

I think in these cases the auto fix should be either disabled (cause such fix feels dangerous) or it should be aware of usage and remove the wrapping brackets.

**Comment by camc314** [comment]
> I think in these cases the auto fix should be either disabled (cause such fix feels dangerous) or it should be aware of usage and remove the wrapping brackets.


Note this is wrong - the fix - wrapping it in `[`, `]` isn't dangerous, and it allocates no more than the previous case.
